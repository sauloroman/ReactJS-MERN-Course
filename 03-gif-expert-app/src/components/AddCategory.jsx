import { useState } from "react"

export const AddCategory = ({ onAddCategory }) => {

  const [category, setCategory] = useState('')

  const onSearchGifs = (e) => {
    e.preventDefault();
    if ( category.trim().length === 0 ) return;

    onAddCategory( category )
    setCategory('')
  }

  return (
    <form data-testid='form-add-category' onSubmit={ onSearchGifs } className="form">
      <div className="form__field">
        <input 
          value={category}
          onChange={ e => setCategory(e.target.value) }
          className="form__input"
          placeholder="Ingrese gifs a buscar"
          type="text" 
        />
        <i className='form__icon bx bx-search-alt-2'></i>
      </div>
    </form>
  )
}
