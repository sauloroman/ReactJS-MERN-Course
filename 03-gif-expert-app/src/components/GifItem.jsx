export const GifItem = ({ gifs }) => {
  return (
    <div className="grid-gifs">
      {
        gifs?.map( ({ id, images: { original: {url}} }) => (
          <img className="grid-gif" key={id} src={ url } alt={ id } />
        ))
      }
    </div>
  )
}
