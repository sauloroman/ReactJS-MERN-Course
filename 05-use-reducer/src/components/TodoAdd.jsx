import { v4 as uuidv4 } from 'uuid'
import { useForm } from '../hooks/useForm'

export const TodoAdd = ({ onAddTodo }) => {

  const { formState, title, description, onInputChange, onResetForm } = useForm({
    title: '',
    description: '',
  })

  const onCreateTodo = (e) => {
    e.preventDefault()

    if ( 
      title.trim().length === 0 || 
      description.trim().length === 0 
    ) return

    formState.id = uuidv4()
    formState.isCompleted = false

    onAddTodo(formState)
    onResetForm()
  }

  return (
    <div className="todo__add">
      <h2 className="todo__title">Agregar nuevo todo</h2>

      <form data-testid='form-todo' onSubmit={ onCreateTodo } action="#" className="todo__form">
        <div className="todo__form-field">
          <label htmlFor="todo-title" className="todo__form-label">Título</label>
          <input
            data-testid='todo-input-title'
            name='title'
            value={title}
            onChange={ onInputChange }
            id='todo-title'
            placeholder="Agrega un título"
            className="todo__form-input" 
            type="text" 
          />
        </div>
        <div className="todo__form-field">
          <label htmlFor="todo-description" className="todo__form-label">Descripción</label>
          <textarea
            data-testid='todo-input-description'
            className='todo__textarea'
            name='description'
            value={description}
            onChange={onInputChange}
            id="todo-description" 
            placeholder='Agrega una descripción '></textarea>
        </div>
        <div className="todo__form-box">
          <button className="todo__form-button">Agregar</button>
        </div>
      </form>

    </div>
  )
}
