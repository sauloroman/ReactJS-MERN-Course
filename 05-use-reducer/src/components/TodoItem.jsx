export const TodoItem = ({ onDeleteTodo, onToggleTodo, id, title, description, isCompleted }) => {
  return (
    <div data-testid="todo" className={`todo ${ isCompleted && 'todo--completed'}`}>
      <div className="todo__info">
        <p className="todo__item-title">{title}</p>
        <p className="todo__description">{description}</p>
      </div>
      <div className="todo__controls">
        <button data-testid="button-toggle" onClick={ () => onToggleTodo( id ) } className="todo__button todo__button--complete">
          { isCompleted ? 'Habilitar' : 'Completar'}
        </button>
        <button data-testid="button-delete" onClick={() => onDeleteTodo( id )} className="todo__button todo__button--delete">Eliminar</button>
      </div>
    </div>
  )
}
