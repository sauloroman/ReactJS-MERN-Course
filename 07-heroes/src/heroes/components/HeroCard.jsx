import { Link } from 'react-router-dom'

const CharactersByHero = ({ alter_ego, characters }) => {
  return ( alter_ego === characters )
  ? <></>
  : <p>{characters}</p>
}

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {

  const heroImageUrl = `/assets/heroes/${id}.jpg`

  return (
    <div className="hero-card">

      <div className="hero-card__side hero-card__front">
        <img className="hero-card__image" src={heroImageUrl} alt={superhero} />
      </div>

      <div className="hero-card__side hero-card__back">
        <header className="hero-card__header">
          <p className="hero-card__name">{superhero}</p>
          <p className="hero-card__publisher">{publisher}</p>
        </header>
        <CharactersByHero alter_ego={alter_ego} characters={characters} />
        
        <div>
          <p className="hero-card__first">Primera aparicion: <span>{first_appearance}</span></p>

          <div className="hero-card__buttonbox">
            <Link className='hero-card__button' to={`/hero/${id}`}>Ver mÃ¡s</Link>
          </div>
        </div>
      </div>
    </div>
  )
}
