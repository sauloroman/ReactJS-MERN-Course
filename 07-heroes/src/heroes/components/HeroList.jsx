import { useMemo } from "react"
import { getHeroesByPublisher } from "../helpers"
import { HeroCard } from "./"

export const HeroList = ({ publisher }) => {

  const heroes = useMemo(() => getHeroesByPublisher( publisher ), [publisher])

  return (
    <div className="hero-grid">
      {
        heroes.map( hero => (
          <HeroCard 
            key={ hero.id }
            { ...hero }
          />
        ))
      }
    </div>
  )
}
