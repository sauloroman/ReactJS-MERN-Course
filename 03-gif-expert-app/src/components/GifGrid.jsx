import { useFetchGifs } from "../hooks/useFetchGifs"
import { GifItem } from "./GifItem"

export const GifGrid = ({ onDeleteCategory, category }) => {

  const { isLoading, data } = useFetchGifs( category )

  return (
    <div className="gifs">
      <header className="gifs__header">
        <h2 className="category-name">{category[0] + category.slice(1).toLowerCase()}</h2>
        <button onClick={ () => onDeleteCategory(category) } className="btn">Eliminar</button>
      </header>
      {
        isLoading
        ? (<p>Cargando...</p>)
        : (<GifItem gifs={data} />)
      }
    </div>
  )
}
