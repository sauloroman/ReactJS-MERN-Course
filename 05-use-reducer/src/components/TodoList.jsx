import { TodoItem } from './TodoItem'

export const TodoList = ({ onDeleteTodo, onToggleTodo, todos }) => {
  return (
    <div className="todos__list">
      <h2 className="todo__title">Todos Creados</h2>

      <div className="todos__list-grid">
        {
          todos?.map( todo => (
            <TodoItem 
              key={todo.id} 
              onDeleteTodo={onDeleteTodo} 
              onToggleTodo={onToggleTodo} 
              {...todo} 
            />
          ))
        }
      </div>
    </div>
  )
}
