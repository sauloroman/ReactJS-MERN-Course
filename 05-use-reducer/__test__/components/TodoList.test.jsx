import { render, screen } from "@testing-library/react"
import { TodoList } from "../../src/components/TodoList"

describe('Tests in TodoList Component', () => {

  const todosMock = [
    {
      id: 'abc',
      title: 'title 1',
      description: 'description 1',
      isCompleted: false,
    },
    {
      id: 'def',
      title: 'title 2',
      description: 'description 2',
      isCompleted: true,
    }
  ]

  const onDeleteTodoMock = jest.fn()
  const onToggleTodoMock = jest.fn()

  test('it must match with snapshot', () => {  
    const { container } = render(<TodoList />)
    expect( container ).toMatchSnapshot()
  })

  test('it should show all todos', () => {

    render(<TodoList 
      onDeleteTodo={onDeleteTodoMock} 
      onToggleTodo={onToggleTodoMock} 
      todos={todosMock}
    />)

    const todos = screen.getAllByTestId('todo')

    expect( todos.length ).toBe(2)

  })
    
}) 