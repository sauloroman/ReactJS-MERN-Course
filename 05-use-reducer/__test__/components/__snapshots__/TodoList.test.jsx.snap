// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in TodoList Component it must match with snapshot 1`] = `
<div>
  <div
    class="todos__list"
  >
    <h2
      class="todo__title"
    >
      Todos Creados
    </h2>
    <div
      class="todos__list-grid"
    />
  </div>
</div>
`;
